### PR TITLE
Add react forms package

### PR DESCRIPTION
**Pull request by ori-shalom**

### CODE REVIEW

File: sdk/react-forms/README.md [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+```tsx
+import { ProtectedForm } from '@piiano/react-form';
+```
+
+## Usage
+
+The following example shows how to use the `ProtectedForm` component to collect credit card information from a user and store it in Piiano Vault collection with the following schema:
+
+```sql
```
Comment (nirhaas) [inline sdk/react-forms/README.md:96]: Is that a pvschema? seems a bit out of context. maybe worth clarifying

File: sdk/react-forms/README.md [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+const fields = [
+  { name: 'card_holder', dataTypeName: 'CC_HOLDER_NAME', label: 'Name', required: true, placeholder: 'John Doe' },
+  { name: 'card_number', dataTypeName: 'CC_NUMBER', label: 'Card', required: true, placeholder: '4111 1111 1111 1111' },
+  { name: 'card_expiry', dataTypeName: 'CC_EXPIRATION_STRING', label: 'Expiry', required: true, placeholder: '12/34' },
+  { name: 'card_cvv', dataTypeName: 'CC_CVV', label: 'CVV', required: true, placeholder: '123' },
+];
+
+function App() {
+  const vaultURL = '';
+  const apiKey = '';
```
Comment (nirhaas) [inline sdk/react-forms/README.md:126]:
```suggestion
  const vaultURL = ''; // Fill this field.
  const apiKey = ''; 	   // Fill this field.
```

File: sdk/react-forms/src/lib/ProtectedForm.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    } else {
+      form.update(options);
+    }
+  }, [options]);
+
+  return { containerRef, form };
+}
+
+type ProtectedFormProps<T extends ResultType> = Omit<ProtectedFormOptions<T>, 'hooks'> &
+  ProtectedFormOptions<T>['hooks'];
```
Comment (nirhaas) [inline sdk/react-forms/src/lib/ProtectedForm.tsx:23]: Are you omitting hooks and then adding it back? what?

File: sdk/react-forms/vite.config.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+async function initDevelopmentVault(): Promise<Vault> {
+  // Start a Vault server to use for testing/development.
+  const vault = new Vault({
+    reuse: true,
+    env: {
+      PVAULT_DEVMODE: true,
+      PVAULT_SENTRY_ENABLE: false,
+      PVAULT_LOG_DATADOG_ENABLE: 'none',
+      // The iframe is loaded from this domain so vault needs to allow it.
```
Comment (nirhaas) [inline sdk/react-forms/vite.config.ts:43]:
```suggestion
      // The iframe is loaded from this domain so Vault needs to allow it.
```

File: sdk/react-forms/index.html [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<!doctype html>
```
[the remark this answers is not in the document]
Comment (ori-shalom) [reply sdk/react-forms/index.html:1]: When running the `yarn dev` it opens the example.
The example files are not packaged to the final library.

File: sdk/react-forms/README.md [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+  - `required` - `boolean` - Whether the field is required or not.
+  - `placeholder` - `string` - The placeholder text for the field.
+  - `value` - `string` - The initial value for the field.
+- `submitButton` - `string` (optional) - The name of the submit button. If provided, a submit button will be rendered. If not provided, the form is expected to be submitted programmatically by the form handler.
+- `style` - `Style` (optional) - Style options to apply to the form.
+  - `theme` - `string` (optional) - The theme to use for the form. Can be one of the following:
+    - `default` - Default theme.
+    - `floating-label` - Floating label theme.
+  - `css` - `string` (optional) - Custom CSS to apply to the form.
+  - `variables` - `Record<string, string>` (optional) - Custom CSS variables to apply to the form. Available variables from the themes are: `primary`, `primaryDark`, `background`, `focusBackground`, `placeholderColor`, and `borderColor`.
```
[the remark this answers is not in the document]
Comment (ori-shalom) [reply sdk/react-forms/README.md:185]: But it's not a flat model.
For example this table is missing documentation for the `Field` type or the `Style` type.

File: sdk/react-forms/src/lib/ProtectedForm.tsx (hunk 3 above)
Comment (ori-shalom) [reply sdk/react-forms/src/lib/ProtectedForm.tsx:23]: The forms options looks like:
`{ vaultURL: '', apiKey: '', ..., hooks: { onSubmit: (value) => {}, onError: (err) => {} } }`
This flatten it to something like this:
`{ vaultURL: '', apiKey: '', ..., onSubmit: (value) => {}, onError: (err) => {} }`
The reason is in react it makes more sense to have these things as direct attributes of the component.

File: sdk/react-forms/index.html (hunk 5 above)
[the remark this answers is not in the document]
Comment (ori-shalom) [reply sdk/react-forms/index.html:1]: Yes but it's not in the src folder either, that's how vite works by default, I can check if it's possible to move it to src/example and configure vite to take it from there.

File: sdk/react-forms/src/lib/ProtectedForm.tsx (hunk 3 above)
Comment (nirhaas) [reply sdk/react-forms/src/lib/ProtectedForm.tsx:23]: ok. Is that a common practice? if not, can you put a comment for smooth brains like me? :)

File: sdk/react-forms/src/example/App.tsx [hunk 7]
```diff
@@ -0,0 +1,43 @@
+import React, { useCallback, useState } from 'react';
+import { createRoot } from 'react-dom/client';
+import './App.css';
+import { ProtectedForm, Result, ResultType } from '../..';
+
+const fields = [
```
Comment (yossi-lavi) [inline sdk/react-forms/src/example/App.tsx:6]: use Fields Type

**Review by yossi-lavi — APPROVED** [review]

**Pull request merged**